zcm/transport/transport_can.cpp

```diff
@@ -133,9 +135,19 @@ struct ZCM_TRANS_CLASSNAME : public zcm_trans_t
     {
         ZCM_TRANS_CLASSNAME* me = cast((zcm_trans_t*) usr);
 
+        uint64_t readStartUtime = TimeUtil::utime();
+
         struct can_frame frame;
         int nbytes = read(me->soc, &frame, sizeof(struct can_frame));
-        if (nbytes != sizeof(struct can_frame)) return 0;
+        if (nbytes != sizeof(struct can_frame)) {
+            // Just sleeping on an error in case we didn't for some reason
```

Comment:
Could you clarify this comment? Who is the "we" in this statement?

Maybe something like "If a timeout is set, make sure we block until the timeout expires on error (otherwise this function could incorrectly get called over and over" or something along those lines.